src/coreclr/jit/fgopt.cpp

```diff
@@ -2346,6 +2339,17 @@ void Compiler::fgCompactBlocks(BasicBlock* block, BasicBlock* bNext)
 
     assert(block->KindIs(bNext->GetKind()));
 
+    if (block->KindIs(BBJ_ALWAYS))
+    {
+        // Propagate BBF_NONE_QUIRK flag
+        block->CopyFlags(bNext, BBF_NONE_QUIRK);
+    }
+    else
+    {
+        // It's no longer a BBJ_ALWAYS; remove the BBF_NONE_QUIRK flag.
+        block->RemoveFlags(BBF_NONE_QUIRK);
```

Comment:
I'd be surprised if removing `BBF_NONE_QUIRK` for blocks that are no longer `BBJ_ALWAYS` produces any diffs, as we should be checking `KindIs(BBJ_ALWAYS` and `JumpsToNext()` first before making any decisions based on `BBF_NONE_QUIRK` being set. Is this change mainly to clean up JIT dumps so `BBF_NONE_QUIRK` isn't set for irrelevant block kinds?

Reply:
> Is this change mainly to clean up JIT dumps so `BBF_NONE_QUIRK` isn't set for irrelevant block kinds?

That's mostly it. I'm also worried about chains of blocks getting compacted such that we end up with a BBJ_ALWAYS with the quirk flag where the block being compacted didn't have it, but I don't have a specific example of that.